k2/csrc/cuda/CMakeLists.txt

```diff
@@ -1,28 +1,15 @@
-add_library(context context.cu)
-target_include_directories(context PUBLIC ${CMAKE_SOURCE_DIR})
-target_link_libraries(context PUBLIC cub)
-target_link_libraries(context PUBLIC glog)
+include_directories(${CMAKE_SOURCE_DIR})
 
-function(k2_add_cuda_test name)
-  add_executable(${name} "${name}.cu")
-  target_link_libraries(${name}
-    PRIVATE
-      context
-      gtest
-      gtest_main
-  )
-  add_test(NAME "Test.Cuda.${name}"
-    COMMAND
-      $<TARGET_FILE:${name}>
-  )
-endfunction()
-
-# please sort the source files alphabetically
-set(cuda_tests
-  ops_test
-  utils_test
+set(CUDA_SOURCE
+    context.cu
 )
 
-foreach(name IN LISTS cuda_tests)
-  k2_add_cuda_test(${name})
-endforeach()
+add_library(k2_cuda ${CUDA_SOURCE})
+target_include_directories(k2_cuda PUBLIC ${CMAKE_SOURCE_DIR})
+target_link_libraries(k2_cuda PUBLIC cub)
+target_link_libraries(k2_cuda PUBLIC glog)
+
+k2_add_gtests(Test.Cuda
+    SOURCES debug_test.cu utils_test.cu #ops_test.cu
```

Comment:
please put the source filenames in a variable, like the above `CUDA_SOURCE`.

Reply:
In this module, cmake_parse_arguments is used, which accepts these into a set. So we don't need to do it out of func.

Reply:
Some time later, this line will contain tens of cuda source filenames, which harms readability.